app/services/course_creation_step_service.rb

```diff
@@ -0,0 +1,139 @@
+class CourseCreationStepService
+  def execute(current_step:, course:)
+    workflow_steps = get_workflow_steps(course)
+
+    {
+      next: get_next_step(workflow_steps, current_step),
+      previous: get_previous_step(workflow_steps, current_step),
+    }
+  end
+
+  def get_next_step(steps, current_step)
+    steps[steps.find_index(current_step).next]
```

Comment:
I find this really hard to parse in my head because of the nesting going on

```
next_index = ...
steps[next_index]
```

Might be a bit nicer